### PR TITLE
Add Debian packaging files @open sesame 3/31 07:59

### PR DESCRIPTION
**Pull request by jijoongmoon**
Add debian packaging files

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Source: nntrainer
+Section: libs
+Priority: optional
+Maintainer: Jijoong Moon <jijoong.moon@samsung.com>
+Build-Depends: gcc-9 | gcc-8 | gcc-7 | gcc-6 | gcc-5 (>=5.4),
+ ninja-build, meson (>=0.50), debhelper (>=9),
+ libblas-dev, libiniparser-dev
+Standards-Version: 0.0.1
```
Comment (wooksong) [inline debian/control]: Standards-Version is not for representing the version of this package. Please refer https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-standards-version

File: debian/control [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Architecture: any
+Multi-Arch: same
+Depends: nntrainer
+Description: NNtrainer development package
+ This is development package for nntrainer.
+
+Package: nntrainer-applications
+Architecture: any
+Multi-Arch: same
+Depends: nntrainer, tesnorflow-dev, jsoncpp, libcurl, ${shlibs:Depends}, ${misc:Depends}
```
Comment (wooksong) [inline debian/control]: Does it depend on tensorflow-dev? In addition, for jsoncpp and libcurl, you just need to add libjsoncpp-dev and libcurl[3|4]-[gnutls|nss|openssl]-dev in Build-Depends section above.

File: debian/control (hunk 1 above)
Comment (jijoongmoon) [reply debian/control]: I see. Thanks.!

File: debian/control (hunk 2 above)
Comment (jijoongmoon) [reply debian/control]: Fixed.! Thanks!

File: debian/control [hunk 3]
```diff
@@ -0,0 +1,32 @@
+Source: nntrainer
+Section: libs
+Priority: optional
+Maintainer: Jijoong Moon <jijoong.moon@samsung.com>
+Build-Depends: gcc-9 | gcc-8 | gcc-7 | gcc-6 | gcc-5 (>=5.4),
+ ninja-build, meson (>=0.50), debhelper (>=9),
+ libblas-dev, libiniparser-dev
```
Comment (wooksong) [inline debian/control]:
```suggestion
Build-Depends: gcc-9 | gcc-8 | gcc-7 | gcc-6 | gcc-5 (>=5.4), pkg-config,
 cmake, ninja-build, meson (>=0.50), debhelper (>=9), libblas-dev,
 libiniparser-dev, tensorflow-lite-dev, libjsoncpp-dev,
 libcurl3-gnutls-dev | libcurl4-gnutls-dev | libcurl3-openssl-dev |
 libcurl4-openssl-dev | libcurl3-nns-dev | libcurl4-nns-dev
```

File: debian/control [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+Multi-Arch: same
+Depends: nntrainer
+Description: NNtrainer development package
+ This is development package for nntrainer.
+
+Package: nntrainer-applications
+Architecture: any
+Multi-Arch: same
+Depends: nntrainer, tesnorflow-lite-dev, libjsoncpp-dev, libcurl[3|4]-[gnutls|nns|openssl]-dev,\
+         ${shlibs:Depends}, ${misc:Depends}
```
Comment (wooksong) [inline debian/control]:
```suggestion
Depends: nntrainer, ${shlibs:Depends}, ${misc:Depends}
```

File: debian/control (hunk 3 above)
Comment (jijoongmoon) [reply debian/control]: done.

File: debian/control (hunk 4 above)
Comment (jijoongmoon) [reply debian/control]: Fixed.

File: debian/nntrainer.install [hunk 5]
```diff
@@ -0,0 +1 @@
+/usr/lib/*/*.so
```
Comment (myungjoo) [inline debian/nntrainer.install]: If you want general applications to use nntrainer, the public .so files should be in libdir, not lib/*/...

Comment (myungjoo) [reply debian/nntrainer.install]: Uh.. sorry. I thought you were talking about /usr/lib/nntrainer/libnntrainer.so
If that was /usr/lib/x86_64.../libnntrainer.so (MULTIARCH), that's fine.

Comment (jijoongmoon) [reply debian/nntrainer.install]: Fixed.

**Pull request merged**